tests/test_sequence.py

```diff
@@ -140,6 +140,200 @@ def test_magnetic_field():
+    )
+
+    test_device3 = Device(
+        name="test_device1",
+        dimensions=2,
+        rydberg_level=60,
+        max_atom_num=100,
+        max_radial_distance=60,
+        min_atom_distance=5,
+        _channels=(
+            (
+                "rmn_local1",
+                Raman.Local(
+                    2 * np.pi * 20,
+                    2 * np.pi * 10,
+                    clock_period=4,
+                    phase_jump_time=750,
+                ),
+            ),
+            (
+                "rmn_local2",
+                Raman.Local(
+                    2 * np.pi * 20,
+                    2 * np.pi * 10,
+                    clock_period=1,
+                    phase_jump_time=500,
+                ),
+            ),
+            (
+                "rmn_local3",
+                Raman.Local(
+                    2 * np.pi * 20,
+                    2 * np.pi * 10,
+                    clock_period=2,
+                    phase_jump_time=500,
+                ),
+            ),
+            (
+                "rydberg_global",
+                Rydberg.Global(
+                    max_abs_detuning=2 * np.pi * 4,
+                    max_amp=2 * np.pi * 3,
+                    phase_jump_time=500,
+                ),
+            ),
+        ),
+    )
+
+    reg = Register.square(3, 5, prefix="q")
+
+    # Device checkout
+    seq = Sequence(reg, Chadoq2)
+    assert seq == seq.switch_device(Chadoq2)
+
+    seq = Sequence(reg, Chadoq2)
+    with pytest.raises(
+        NotImplementedError,
+        match="Switching the device of a sequence"
+        + " to 'MockDevice' is not supported.",
+    ):
+        seq.switch_device(MockDevice)
+
+    # Channel basis
+    seq = Sequence(reg, test_device1)
+    seq.declare_channel("digital", "raman_global", "q1")
+
+    with pytest.raises(
+        TypeError, match="No channel with both basis and addressing match."
+    ):
+        seq.switch_device(IroiseMVP)
+
+    # Channel addressing
+    seq = Sequence(reg, test_device1)
+    seq.declare_channel("ising", "raman_global")
+
+    with pytest.raises(
+        TypeError, match="No channel with both basis and addressing match."
+    ):
+        seq.switch_device(test_device2)
+
+    seq = Sequence(reg, MockDevice)
+    seq.declare_channel("ising", "raman_global")
+    new_seq = seq.switch_device(test_device1)
+    assert new_seq.declared_channels["ising"].basis == "digital"
+    assert new_seq.declared_channels["ising"].addressing == "Global"
+
+    # Channel addressing and basis issue
+    seq = Sequence(reg, test_device2)
+    seq.declare_channel("ising", "rmn_local")
+
+    with pytest.raises(
+        TypeError, match="No channel with both basis and addressing match."
+    ):
+        seq.switch_device(IroiseMVP)
+
+    # Strict-Jump_phase_time & CLock-period
+    # Jump_phase_time: with post_phase_time
+    R_interatomic = Chadoq2.rydberg_blockade_radius(2 * np.pi)
+    reg = Register.square(3, R_interatomic, prefix="q")
```

Comment:
nit: Let's stick to python style conventions and not use an uppercase letter for a variable
```suggestion
    r_interatomic = Chadoq2.rydberg_blockade_radius(2 * np.pi)
    reg = Register.square(3, r_interatomic, prefix="q")
```